src/test/java/io/vertx/ext/asyncsql/SQLTestBase.java

```diff
@@ -0,0 +1,490 @@
+          ensureSuccess(context, ar3);
+          conn.updateWithParams("UPDATE test_table SET name=? WHERE id=?",
+              new JsonArray().add(name).add(id),
+              ar4 -> {
+                ensureSuccess(context, ar4);
+                UpdateResult updateRes = ar4.result();
+                conn.rollback(ar5 -> {
+                  ensureSuccess(context, ar5);
+                  conn.query("SELECT name FROM test_table ORDER BY id", ar6 -> {
+                    ensureSuccess(context, ar6);
+                    ResultSet selectRes = ar6.result();
+                    context.assertNotNull(updateRes);
+                    context.assertNotNull(selectRes);
+                    context.assertEquals(1, updateRes.getUpdated());
+                    context.assertEquals("Albert",
+                        selectRes.getResults().get(0).getString(0));
+                    async.complete();
+                  });
+                });
+              });
+        });
+      });
+    });
+  }
+
+  @Test
+  public void testMultipleConnections(TestContext context) {
+    int id = 0;
+    String name = "Adele";
+
+    SQLConnection[] connections = new SQLConnection[2];
+
+    Async async1 = context.async();
+    Async async2 = context.async();
+    client.getConnection(ar -> {
+      ensureSuccess(context, ar);
+      conn = ar.result();
+      setupSimpleTable(conn, ar2 -> {
+        client.getConnection(ar3 -> {
+          ensureSuccess(context, ar3);
+          connections[0] = ar3.result();
+          connections[0].setAutoCommit(false, ar31 -> {
+            ensureSuccess(context, ar31);
+            connections[0].updateWithParams("UPDATE test_table SET name=? WHERE id=?",
+                new JsonArray().add(name).add(id), ar32 -> {
+                  ensureSuccess(context, ar32);
+                  context.assertEquals(ar32.result().getUpdated(), 1);
+                  connections[0].rollback(ar33 -> {
+                    ensureSuccess(context, ar33);
+                    connections[0].close(v -> {
+                      ensureSuccess(context, v);
+                      async1.complete();
+                    });
+                  });
+                });
+          });
+        });
+        client.getConnection(ar4 -> {
+          ensureSuccess(context, ar4);
+          connections[1] = ar4.result();
+          connections[1].query("SELECT name FROM test_table ORDER BY id", ar41 -> {
+            ensureSuccess(context, ar41);
+            ResultSet resultSet = ar41.result();
+            context.assertEquals(resultSet.getResults().get(0).getString(0), "Albert");
+            connections[1].close(v -> {
+              ensureSuccess(context, v);
+              async2.complete();
+            });
+          });
+        });
+      });
+    });
+  }
+
+  @Test
+  public void testSetAutocommitWhileInTransaction(TestContext context) {
+    //TODO not sur it is the correct translation, need to be checked
```

Comment:
From the logic, it looks like the Scala version:

Get a connection, set auto commit to `false`, do something with it, set it to `true` again and check with a second connection that the update really took place.